merino/providers/accuweather.py

```diff
@@ -0,0 +1,154 @@
+"""AccuWeather integration."""
+import logging
+import os
+from typing import Any, Optional
+from urllib.parse import urlencode, urlunparse
+
+import httpx
+from fastapi import FastAPI, Request
+
+from merino.config import settings
+from merino.providers.base import BaseProvider, BaseSuggestion
+
+logger = logging.getLogger(__name__)
+
+
+def get_value(d: dict, keys: list, default_value: Any = None) -> Any:
+    """Get a nested value in a sequence of dictionaries."""
+    for k in keys:
+        if d is None or not isinstance(d, dict):
+            return default_value
+        d = d.get(k)
+    return d
+
+
+class Suggestion(BaseSuggestion):
+    """Model for AccuWeather suggestions."""
+
+    city_name: Optional[str] = None
+    temperature_unit: Optional[str] = None
+    high: Optional[float] = None
+    low: Optional[float] = None
+    day_summary: Optional[str] = None
+    day_precipitation: Optional[bool] = None
+    night_summary: Optional[str] = None
+    night_precipitation: Optional[bool] = None
+
+
+class Provider(BaseProvider):
+    """Suggestion provider for AccuWeather."""
+
+    _app: FastAPI
+
+    def __init__(
+        self, app: FastAPI = None, enabled_by_default: bool = False, **kwargs: Any
+    ) -> None:
+        self._app = app
+        self._enabled_by_default = enabled_by_default
+        super().__init__(**kwargs)
+
+    async def initialize(self) -> None:
+        """Initialize the provider."""
+        ...
+
+    def hidden(self) -> bool:  # noqa: D102
+        return False
+
+    async def handle_request(self, request: Request) -> list[BaseSuggestion]:
+        """Provide suggestions for a given request."""
+        api_key = os.environ.get("MERINO_ACCUWEATHER_API_KEY")
+        if api_key is None:
+            logger.warning("AccuWeather API key not specified")
+            return []
+
+        country = None
+        postal_code = None
+        try:
+            country = request.state.location.country
+            postal_code = request.state.location.postal_code
```

Comment:
As part of the middleware rewrite, we no longer store the location data inside of `request.state`. Instead we use a `ContextVar` to store that for each request. Here is an [example](https://github.com/mozilla-services/merino-py/blob/main/merino/middleware/logging.py#L41).

```suggestion
            # need to import the location ContextVar
            from merino.middleware.geolocation import ctxvar_geolocation
            
            # `try except` is not needed here
            location = ctxvar_geolocation.get()
            country = location.country
            postal_code = location.postal_code
            if country is None or postal_code is None:
                 logger.warning("Country and/or postal codes unknown")
                 return []
```

Reply:
I see. That means my commit where I replaced `query()` with `handle_request()` isn't necessary at all, nice.

Reply:
I ended up leaving my `handle_request()` change because I think it still makes sense when some providers depend on certain query params and others don't. I'm not sure if that's the best way to handle it, let me know if not.